hudi-spark-datasource/hudi-spark3/src/main/scala/org/apache/spark/sql/hudi/catalog/HoodieCatalog.scala

```diff
@@ -105,12 +106,16 @@ class HoodieCatalog extends DelegatingCatalogExtension
           case _ =>
             catalogTable0
         }
-        HoodieInternalV2Table(
+
+        val v2Table = HoodieInternalV2Table(
           spark = spark,
           path = catalogTable.location.toString,
           catalogTable = Some(catalogTable),
           tableIdentifier = Some(ident.toString))
-      case o => o
+        // TODO elaborate
+        v2Table.v1TableWrapper
```

Comment:
it means users no long able to write/read data when specifying HoodieCatalog

Reply:
Why? Catalog still exposes the methods to write into Hudi tables